feature/system/gnmi/set/tests/gnmi_set_test/gnmi_set_test.go

```diff
@@ -695,21 +770,23 @@ func verifyAggregate(t testing.TB, dev gnmi.DeviceOrOpts, aggID string, a *attrs
 func verifyInterface(t testing.TB, dev gnmi.DeviceOrOpts, name string, a *attrs.Attributes) {
 	t.Helper()
 	q := gnmi.OC().Interface(name).Subinterface(0).Ipv4().Address(a.IPv4).PrefixLength().State()
-	v := gnmi.Lookup(t, dev, q)
-	if got, ok := v.Val(); !ok || got != a.IPv4Len {
+	v, ok := gnmi.Await(t, dev, q, 60*time.Second, a.IPv4Len).Val()
+	if !ok {
 		t.Errorf("State got %v, want %v", v, a.IPv4Len)
 	} else {
 		t.Logf("Verified %v", v)
 	}
 }
 
 // attachInterface attaches an interface name and subinterface sub to a network instance.
-func attachInterface(ni *oc.NetworkInstance, name string, sub int) string {
+func attachInterface(dut *ondatra.DUTDevice, ni *oc.NetworkInstance, name string, sub int) string {
 	id := name // Possibly vendor specific?  May have to use sub.
 	niface := ni.GetOrCreateInterface(id)
 	niface.Interface = ygot.String(name)
-	niface.Subinterface = ygot.Uint32(uint32(sub))
-	id = fmt.Sprintf("%s.%d", id, sub)
+	if !deviations.SkipMainInterfaceDotZero(dut) {
+		niface.Subinterface = ygot.Uint32(uint32(sub))
+		id = fmt.Sprintf("%s.%d", id, sub)
```

Comment:
While this is not introduced in this PR, this id change seems incorrect and introduces an inconsistent usage interface IDs throughout the test. Some gnmi calls use "ethernet-x" and  others "ethernet-x.y". Not sure if these inconsistencies are directly related to other changes in this PR (like changes of the scope of variables).

Could you please put this single line (`id = fmt.Sprintf("%s.%d", id, sub)` under deviation `InterfaceRefInterfaceIDFormat`)? thanks

cc @cfernanz